recurring_contract/model/recurring_contract.py

```diff
@@ -291,6 +326,20 @@ def clean_invoices(self, cr, uid, ids, context=None, since_date=None,
             wf_service.trg_validate(uid, 'account.invoice',
                                     inv.id, 'invoice_open', cr)
 
+        for contract in self.browse(cr, uid, ids, context):
+            next_invoice_date = since_date
+
+            if contract.last_paid_invoice_date:
+                next_invoice_date = max(
+                    datetime.strptime(
+                        contract.last_paid_invoice_date, DF), since_date)
+
+            self.write(

```

Comment:
I see two problematic use cases with this addition :

* Suspension : it will call clean_invoices to avoid paying in the period of suspension, but the next_invoice_date should not be updated !
* Contract termination : it will call clean_invoices but next_invoice_date does not not need to be updated.

You should avoid this either by adding a value in the context for preventing the modification of next_invoice_date, or adding a parameter to the method. You can then search all the calls to this method and see where it makes sense to update next_invoice_date and where it doesn't make sense to do it.